### PR TITLE
CEM-1285 Vendors-Header-Molecule

### PR DESCRIPTION
**Pull request by xsharonhe**
Screen Shot 2020-12-05 at 12 07 54 AM

### CODE REVIEW

File: src/Containers/VendorsList/VendorsHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    return (
+        <div {...props}>
+            <Heading type="h1" bold>
+                {headerName}
+            </Heading>
+            <div style={{ ...rowStyle }}>
+                <Button
+                    icon={icon}
+                    onClick={onImportButtonClick}
+                    style={{ ...buttonStyle }}
```
Comment (ralph-dev) [inline src/Containers/VendorsList/VendorsHeader.tsx:41]: leftButtonProps, what if user wants to disable the button?

File: src/Containers/VendorsList/VendorsHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                />
+                <Button 
+                    icon={buttonIcon} 
+                    onClick={onAddButtonClick} 
+                    primary
```
Comment (ralph-dev) [inline src/Containers/VendorsList/VendorsHeader.tsx:46]: rightButtonProps

File: stories/Containers/VendorsHeader.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import { VendorsHeader, VendorsHeaderProps } from '../../src';
+import { createStoryTitle } from '../Constants';
+import { Story, Meta } from '@storybook/react';
+
+export default {
+    title: createStoryTitle('Vendors Header'),
+    component: VendorsHeader,
+} as Meta;
+
+export const defaultArgs = {
```
Comment (xsharonhe) [inline stories/Containers/VendorsHeader.stories.tsx:13]: You mentioned defaultArgs in the call - not sure if this is exactly the proper way?

File: src/Containers/VendorsList/VendorsHeader.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import { Heading } from '../../Text/Heading';
+import { MainInterface, ResponsiveInterface } from '../../Utils/BaseStyles';
+import { flex } from '../../Utils/Mixins';
+
+export interface VendorsHeaderProps
+    extends MainInterface,
+        ResponsiveInterface,
+        React.HTMLAttributes<HTMLDivElement> {
+    headerName: string;
+    buttonName: string;
```
Comment (ralph-dev) [inline src/Containers/VendorsList/VendorsHeader.tsx:13]: which button name? right button name?

Comment (ralph-dev) [reply src/Containers/VendorsList/VendorsHeader.tsx:13]: Also i would say rightButtonText lol

File: src/Containers/VendorsList/VendorsHeader.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+import { Button, ButtonProps } from '../../Inputs/Button';
+import { Heading } from '../../Text/Heading';
+import { MainInterface, ResponsiveInterface } from '../../Utils/BaseStyles';
+import { flex } from '../../Utils/Mixins';
+
+export interface VendorsHeaderProps
+    extends MainInterface,
+        ResponsiveInterface,
+        React.HTMLAttributes<HTMLDivElement> {
+    headerName: string;
```
Comment (ralph-dev) [inline src/Containers/VendorsList/VendorsHeader.tsx:12]: headerText

File: stories/Containers/VendorsHeader.stories.tsx (hunk 3 above)
Comment (ralph-dev) [reply stories/Containers/VendorsHeader.stories.tsx:13]: defaultArgs will collide with other defaultArgs make it unique and easy to use.
Ex: getVendorHeaderProps

Comment (xsharonhe) [reply stories/Containers/VendorsHeader.stories.tsx]: hopefully this change is better?

File: src/Containers/VendorsList/VendorsHeader.tsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+import { MainInterface, ResponsiveInterface } from '../../Utils/BaseStyles';
+import { flex } from '../../Utils/Mixins';
+
+export interface VendorsHeaderProps
+    extends MainInterface,
+        ResponsiveInterface,
+        React.HTMLAttributes<HTMLDivElement> {
+    headerText: string;
+    rightButtonText: string;
+    leftButtonProps: ButtonProps;
```
Comment (ralph-dev) [inline src/Containers/VendorsList/VendorsHeader.tsx:14]: Btw I like to preface all my interfaces with I to make it eaiser to auto correct

**Review by ralph-dev — APPROVED** [review]
LGTM can you just make sure the controls are working? I don't if you still have access to controls btw, you should.

**Review by ralph-dev — APPROVED** [review]
Very well done!

**Pull request merged**